Index: setup.py
===================================================================
--- setup.py	(revision 875)
+++ setup.py	(working copy)
@@ -264,7 +264,8 @@
             L.debug("Note: the path required is to where the static library lib is found")
             self.with_BLAS = False
         else:
-            if not os.path.exists(os.path.join(self.BLASdir,self.BLASname_t+'.a')):
+            shlib_ext = {'linux': '.so', 'mac': '.dylib', 'win': '.lib'}.get(self.platform, '')
+            if not os.path.exists(os.path.join(self.BLASdir,self.BLASname_t+'.a')) and not os.path.exists(os.path.join(self.BLASdir,self.BLASname_t+shlib_ext)):
                 L.warning("Could not find BLAS"+msg)
                 L.debug("Could not find BLAS at the given path {}.".format(self.BLASdir))
                 L.debug("usage: --blas-home=path")
