diff --git a/setup.py b/setup.py
index 2e45d83..adee161 100644
--- a/setup.py
+++ b/setup.py
@@ -553,13 +553,16 @@ class Assimulo_prepare(object):
         Adds the Fortran extensions using Numpy's distutils extension.
         """
         extra_link_flags = self.static_link_gfortran + self.static_link_gcc + self.flag_32bit + self.extra_fortran_link_flags
+        L.debug("fortran_extensionlists extra_link_flags=" + str(extra_link_flags))
         extra_compile_flags = self.flag_32bit + self.extra_c_flags + self.extra_fortran_compile_flags
-        
+        L.debug("fortran_extensionlists extra_compile_flags=" + str(extra_compile_flags))
+
         config = np.distutils.misc_util.Configuration()
         extraargs={'extra_link_args':extra_link_flags[:], 'extra_compile_args':extra_compile_flags[:]}
                   
         extraargs['extra_f77_compile_args'] = extra_compile_flags[:]
         extraargs['extra_f90_compile_args'] = extra_compile_flags[:]
+        L.debug("fortran_extensionlists extraargs=" + str(extraargs))
     
         #Hairer
         sources='assimulo'+os.sep+'thirdparty'+os.sep+'hairer'+os.sep+'{0}.f','assimulo'+os.sep+'thirdparty'+os.sep+'hairer'+os.sep+'{0}.pyf'
@@ -599,6 +602,7 @@ class Assimulo_prepare(object):
         if self.with_BLAS and self.with_LAPACK:
             extraargs_glimda={'extra_link_args':extra_link_flags[:], 'extra_compile_args':extra_compile_flags[:], 'library_dirs':[self.BLASdir, self.LAPACKdir], 'libraries':['lapack', self.BLASname]}
             extraargs_glimda["extra_f77_compile_args"] = extra_compile_flags[:]
+            L.debug("fortran_extensionlists extraargs_glimda=" + str(extraargs_glimda))
             config.add_extension('assimulo.lib.glimda', sources= src,include_dirs=[np.get_include()],**extraargs_glimda) 
             extra_link_flags=extra_link_flags[:-2]  # remove LAPACK flags after GLIMDA
         elif self.with_MKL: #assuming windows and Intel fortran compiler
