diff --git a/setup.py b/setup.py
index db510cb..d8f0d2b 100644
--- a/setup.py
+++ b/setup.py
@@ -361,7 +361,9 @@ class Assimulo_prepare(object):
                     
             #if self.with_BLAS:
             #    self.superLUFiles.append(self.BLASname)
-            
+            if "win" in self.platform:
+                self.superLUFiles.append("blas.lib")
+
             L.debug("SuperLU files: "+str(self.superLUFiles))
             
         else:
@@ -538,11 +540,11 @@ class Assimulo_prepare(object):
         if self.with_SLU:
             if 'win' in self.platform:
                 ext_list[-1].library_dirs = [os.path.join(self.SLUincdir, "..", "lib"), self.libdirs]
-                ext_list[-1].libraries = ['superlu_mt_OPENMP', 'blas_OPENMP']
-                ext_list[-1].extra_compile_args += ["-D__OPENMP", "-D__RADAU5_WITH_SUPERLU"]
+                ext_list[-1].libraries = ['superlu_mt_OPENMP', 'blas']
+                ext_list[-1].extra_compile_args += ["/D__OPENMP", "/D__RADAU5_WITH_SUPERLU"]
             else:
                 ext_list[-1].library_dirs = [os.path.join(self.SLUincdir, "..", "lib"), self.BLASdir]
-                ext_list[-1].libraries = ['superlu_mt_OPENMP', 'blas_OPENMP', 'blas', 'm', 'gomp']
+                ext_list[-1].libraries = ['superlu_mt_OPENMP', 'blas', 'm', 'gomp']
                 ext_list[-1].extra_compile_args = ["-D__RADAU5_WITH_SUPERLU"]
         else:
             if 'win' not in self.platform:
@@ -571,6 +573,9 @@ class Assimulo_prepare(object):
                     openmp_arg = "-fopenmp"
                 el.extra_link_args.append(openmp_arg)
                 el.extra_compile_args.append(openmp_arg)
+            if self.with_SLU:
+                if "win" in self.platform:
+                    el.extra_compile_args += ["/D__OPENMP"]
             el.extra_compile_args += self.flag_32bit + self.extra_c_flags
 
         for el in ext_list:
