diff --git a/setup.py b/setup.py
index b98a8bd..d52a777 100644
--- a/setup.py
+++ b/setup.py
@@ -542,7 +542,7 @@ class Assimulo_prepare(object):
                 ext_list[-1].extra_compile_args += ["-D__OPENMP", "-D__RADAU5_WITH_SUPERLU"]
             else:
                 ext_list[-1].library_dirs = [os.path.join(self.SLUincdir, "..", "lib"), self.BLASdir]
-                ext_list[-1].libraries = ['superlu_mt_OPENMP', 'blas_OPENMP', 'blas', 'm', 'gomp']
+                ext_list[-1].libraries = ['superlu_mt_OPENMP', 'blas', 'm', 'gomp']
                 ext_list[-1].extra_compile_args = ["-D__RADAU5_WITH_SUPERLU"]
         else:
             if 'win' not in self.platform:
